functions/tips_retry/index.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_tips_retry]
+
+use Google\CloudFunctions\CloudEvent;
+
+function tipsRetry(CloudEvent $event): string
+{
+    $data = $event->getData()['data'];
+    $data = json_decode(base64_decode($data), true);
+
+    // Determine whether to retry the invocation based on a parameter
+    $tryAgain = $data['some_parameter'];
+
+    if ($tryAgain) {
+        /**
+         * Functions with automatic retries enabled should throw exceptions to
+         * indicate temporary failures that a retry might fix. In this case,
+         * a thrown exception will cause the original function invocation to be
+         * re-sent.
+         */
+        throw new Exception('Retrying...');
```

Comment:
nit: While "Retrying..." here is the result of sending the exception, all the function knows is:

1. Something went wrong
2. We want a retry to happen.

I'm unsure if the goal of this snippet should be hyper-focused on training retry expectations, or if we should also include the angle of making the retry semantics friendly to troubleshooting. The exception ideally includes context on why the retry is needed so Cloud Logging and Cloud Error Reporting can provide more details:

```suggestion
        throw new Exception('Try Again triggered, expecting retry...');
```